lib/ansible/modules/cloud/amazon/lambda_policy.py

```diff
@@ -0,0 +1,529 @@
+    qualifier = get_qualifier(module)
+    if qualifier:
+        api_params.update(Qualifier=qualifier)
+
+    try:
+        if not module.check_mode:
+            client.remove_permission(**api_params)
+        changed = True
+    except Exception as e:
+        fail_json_aws(module, e, msg="removing permission from policy")
+
+    return changed
+
+
+def manage_state(module, aws):
+    changed = False
+    current_state = 'absent'
+    state = module.params['state']
+    action_taken = 'none'
+
+    # check if the policy exists
+    current_policy_statement = get_policy_statement(module, aws)
+    if current_policy_statement:
+        current_state = 'present'
+
+    if state == 'present':
+        if current_state == 'present':
+            # check if policy has changed and update if necessary
+            # since there's no API to update a policy statement, it must first be removed
+            if not policy_equal(module, current_policy_statement):
+                remove_policy_permission(module, aws)
+                changed = add_policy_permission(module, aws)
+                action_taken = 'updated'
+        else:
+            # add policy statement
+            changed = add_policy_permission(module, aws)
+            action_taken = 'added'
+    else:
+        if current_state == 'present':
+            # remove the policy statement
+            changed = remove_policy_permission(module, aws)
+            action_taken = 'deleted'
+
+    return dict(changed=changed, ansible_facts=dict(lambda_policy_action=action_taken))
+
+
+# ---------------------------------------------------------------------------------------------------
+#
+#   MAIN
+#
+# ---------------------------------------------------------------------------------------------------
+
+def main():
+    """
+    Main entry point.
+
+    :return dict: ansible facts
+    """
+
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            state=dict(default='present', choices=['present', 'absent']),
+            function_name=dict(required=True, aliases=['lambda_function_arn', 'function_arn']),
+            statement_id=dict(required=True, aliases=['sid']),
+            alias=dict(),
+            version=dict(type='int', default=0),
+            action=dict(required=True, ),
+            principal=dict(required=True, ),
+            source_arn=dict(),
+            source_account=dict(),
+            event_source_token=dict(),
+        )
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        mutually_exclusive=[['alias', 'version'],
+                            ['event_source_token', 'source_arn'],
+                            ['event_source_token', 'source_account']],
+    )
+
+    # validate dependencies
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto3 and botocore are required for this module.')
+
+    try:
+        aws = AWSConnection(module, ['lambda'])
```

Comment:
Rather than pass aws everywhere we could just pass the connection and simplify a little?